src/coreclr/jit/assertionprop.cpp

```diff
@@ -6453,6 +6590,17 @@ PhaseStatus Compiler::optAssertionPropMain()
         }
     }
 
+    if (containsSwitches)
+    {
+        for (BasicBlock* const block : Blocks())
```

Comment:
There are unlikely to be many switches so maybe keep track of them in the initial walk?

Reply:
Done